File: backend/AuthenticationSystem/src/main/java/com/vbt/AuthenticationSystem/infrastructure/exception/ApiErrorResponse.java
package com.vbt.AuthenticationSystem.infrastructure.exception;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.time.Instant;
import java.time.LocalDateTime;
import java.util.Map;

// Sadece dolu olan (null olmayan) alanlar JSON'a dahil edilsin
@JsonInclude(JsonInclude.Include.NON_NULL)
public record ApiErrorResponse(
        String title,
        int status,
        String detail,
        String instance,
        Instant timestamp,
        Map<String, String> invalidParams // Sadece form validasyon hataları olduğunda dolacak
) {
}

File: mobile/lib/features/authentication/data/models/api_error_response.dart
class ApiErrorResponse {
  final String title;
  final int status;
  final String detail;
  final String instance;
  final String timestamp;
  final Map<String, dynamic>? invalidParams;

  const ApiErrorResponse({
    required this.title,
    required this.status,
    required this.detail,
    required this.instance,
    required this.timestamp,
    this.invalidParams,
  });

  factory ApiErrorResponse.fromJson(Map<String, dynamic> json) {
    return ApiErrorResponse(
      title: json["title"] ?? "",
      status: json["status"] ?? 0,
      detail: json["detail"] ?? "",
      instance: json["instance"] ?? "",
      timestamp: json["timestamp"] ?? "",
      invalidParams: json["invalidParams"],
    );
  }
}